tests/ParserTests/TestFileReader.php

```
@@ -48,7 +48,7 @@ private function __construct(
 			} elseif ( $item['type'] === 'comment' ) {
 				$lastComment .= $item['text'];
 			} elseif ( $item['type'] === 'line' ) {
-				if ( !empty( trim( $item['text'] ) ) ) {
+				if ( trim( $item['text'] ) != false ) {
```

Shouldn't this be `!== ''` ? Trim should never return false, it returns string.

I think this should probably actually read:

`if ( trim ( $item['text'] ?? '' ) !== '' ) {`

That is -- the `empty` was one of those magic PHP functions that swallows up "undefined property" errors, so we should preserve that by using the `??` operator.  There are probably lots of other places we could use `??` in a PHP 8 world as well.